File: server/src/main/java/com/khanabook/saas/sync/dto/PushSyncResponse.java
package com.khanabook.saas.sync.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PushSyncResponse {
    private List<Integer> successfulLocalIds;
    private List<Integer> failedLocalIds;
}

File: Android/app/src/main/java/com/khanabook/lite/pos/data/remote/dto/PushSyncResponse.kt
package com.khanabook.lite.pos.data.remote.dto

import com.google.gson.annotations.SerializedName

/**
 * Data class matching the server's PushSyncResponse shape.
 * Used to handle sync responses without crashing.
 */
data class PushSyncResponse(
    @SerializedName("successfulLocalIds")
    val successfulLocalIds: List<Int>,
    @SerializedName("failedLocalIds")
    val failedLocalIds: List<Int>
)
